<launch>

    <arg name="sim" default="false"/>
    <arg name="no_remote" default="false"/>

    <group unless="$(arg sim)">
        <param name="/lidar/zero_position" type="double" value="3.5"/>
        <param name="/lidar/speed" type="double" value="9.0"/>
        <param name="/lidar/frame_id" type="str" value="laserscanner_frame"/>

        <group if="$(arg no_remote)">
            <node pkg="ros_lidar" type="lidar_laserscan_publisher.py" name="lidar_laserscan_publisher" args="lidar" required="true">
                <remap from="/lidar/port_lidar_sensor" to="/peripherials_ports/lidar_sensor" />
                <remap from="/lidar/port_motor" to="/peripherials_ports/arduino_lidar" />
            </node>
        </group>

        <group unless="$(arg no_remote)">
            <machine
                name="robot"
                address="ddbot"
                user="ubuntu"
                env-loader="/home/ubuntu/setup_ros.bash"
                default="never"
            />
            <node pkg="ros_lidar" type="lidar_laserscan_publisher.py" name="lidar_laserscan_publisher" args="lidar" required="true" machine="robot">
                <remap from="/lidar/port_lidar_sensor" to="/peripherials_ports/lidar_sensor" />
                <remap from="/lidar/port_motor" to="/peripherials_ports/arduino_lidar" />
            </node>
        </group>
    </group>

</launch>